src/app/core/auth/server-auth.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+
+import { Observable } from 'rxjs/Observable';
+import { HttpHeaders } from '@angular/common/http';
+import { HttpOptions } from '../dspace-rest-v2/dspace-rest-v2.service';
+import { AuthStatus } from './models/auth-status.model';
+import { isNotEmpty } from '../../shared/empty.util';
+import { AuthService } from './auth.service';
+import { AuthTokenInfo } from './models/auth-token-info.model';
+import { CheckAuthenticationTokenAction } from './auth.actions';
+
+/**
+ * The auth service.
+ */
+@Injectable()
+export class ServerAuthService extends AuthService {
+
+  /**
+   * Authenticate the user
+   *
+   * @param {string} user The user name
+   * @param {string} password The user's password
+   * @returns {Observable<User>} The authenticated user observable.
+   */
+  public authenticate(user: string, password: string): Observable<AuthStatus> {
+    // Attempt authenticating the user using the supplied credentials.
+    const body = encodeURI(`password=${password}&user=${user}`);
+    const options: HttpOptions = Object.create({});
+    let headers = new HttpHeaders();
+
+    // NB this could be use to avoid the problem with the authentication is case the UI is rendered by Angular Universal.
+    const clientIp = this.req.connection.remoteAddress;
+
+    headers = headers.append('Content-Type', 'application/x-www-form-urlencoded');
```

Comment:
Could you also add 
```
headers = headers.append('X-Forwarded-For', clientIp);
``` 
here? That can already help solving the "disable ip check" problem in the REST backend.

Reply:
@tomdesair 

done